lib/IRGen/GenStruct.cpp

```diff
@@ -290,12 +317,27 @@ namespace {
       }
     }
   };
-  
+
   /// A type implementation for loadable record types imported from Clang.
   class ClangRecordTypeInfo final :
     public StructTypeInfoBase<ClangRecordTypeInfo, LoadableTypeInfo,
                               ClangFieldInfo> {
     const clang::RecordDecl *ClangDecl;
+
+    /// Try to emit a call of the destructor of \p address. If there's no
+    /// destructor, return false. Otherwise, create the call and return true.
+    static bool callCXXDestructor(IRGenFunction &IGF, Address address, SILType type) {
```

Comment:
Again, I'm not sure it's worth putting this in a separate function.

If you inline this into `destroy`, the only added complexity is the call to the base class's `destroy` instead of doing a `return false` -- but you get the benefit of eliminating another function to think about.

Reply:
Will do.